HousingManagementSystemApi.Tests/ContollersTests/AddressControllerTests.cs

```diff
@@ -71,7 +72,7 @@ public async Task GivenAnExceptionIsThrown_WhenRequestMadeForAddresses_ResponseH
             // Arrange
             const string errorMessage = "An error message";
             retrieveAddressesUseCaseMock.Setup(x => x.Execute(It.IsAny<string>()))
-                .Throws(new Exception(errorMessage));
+                .Throws(new ConfigurationErrorsException(errorMessage));
```

Comment:
Is this really a configuration error?
Given the system under test, AddresssController, catches `Exception` ([here](https://github.com/Newarkandsherwood/HousingManagementSystemApi/blob/main/HousingManagementSystemApi/Controllers/AddressesController.cs#L28)), I feel it's appropriate to use the same exception here.
This test doesn't cover the scenario where the use case throws a different exception and ensures that exception is handling by AddressController.

Given this warning existed prior to any of these changes and no changes to the AddressController were made that resulted in the warning, I suggest restoring the test to use `Exception`.

The (pre-existing) CI warning should be tackled independently.

`ConfigurationErrorsException` would be ideal to use when our configuration isn't setup correctly, e.g. [EnvironmentVariableHelper](https://github.com/Newarkandsherwood/housing-repairs-online-api/blob/09d74ab082b600bce8eaab26fcd006c58abeeb63/HousingRepairsOnlineApi/Helpers/EnvironmentVariableHelper.cs#L10)

Reply:
I was going through trying to get rid of warnings. You're right, though - this was pre existing and whilst this change got rid of the warning, it's best left for now.